src/org/rascalmpl/library/Prelude.java

```diff
@@ -1483,18 +1486,38 @@ public IString readBase64(ISourceLocation sloc) {
         }
     }
 
-	public void writeBase64(ISourceLocation sloc, IString contents) {
+	public void writeBase64(ISourceLocation sloc, IString base64content) {
         int BUFFER_SIZE = 3 * 512;
         Base64.Decoder decoder = Base64.getDecoder();
         
         try  (BufferedOutputStream out = new BufferedOutputStream(REGISTRY.getOutputStream(sloc, false), BUFFER_SIZE); ) {
-			out.write(decoder.decode(contents.getValue()));
+			out.write(decoder.decode(base64content.getValue()));
         }
         catch (IOException e) {
             throw RuntimeExceptionFactory.io(values.string(e.getMessage()));
         }
     }
 	
+	public IString readBase32(ISourceLocation sloc) {
+		try(BufferedInputStream input = new BufferedInputStream(REGISTRY.getInputStream(sloc))) {
+			Base32 encoder = new Base32();
+			String encoded = encoder.encodeToString(input.readAllBytes());
+			return values.string(encoded);
+		} catch (IOException e) {
+            throw RuntimeExceptionFactory.io(values.string(e.getMessage()));
+		}
+    }
+
+	public void writeBase32(ISourceLocation sloc, IString base32Content) {
+        try  (BufferedOutputStream output = new BufferedOutputStream(REGISTRY.getOutputStream(sloc, false))) {
+			Base32 decoder = new Base32(72, new byte[] { '\r', '\n' }, false, (byte) '=', CodecPolicy.LENIENT);
```

Comment:
should the newline be hardcoded to windows style? if so, can we document it in the code that it's because of an RFC or something?

Reply:
We only want to set the policy to LENIENT, but the only way to do that is specify the other parameters as well. So I just used the default values that would be used if you do not specify any parameters in the constructor.
The main issue is that `Base32.Builder` is missing somehow missing from the commons-codec jar, otherwise we could have used that and only override the policy setting.
I will add a comment in the code to explain this.

Reply:
So even on linux we'll get `\r\n`?

Reply:
Indeed, see: https://commons.apache.org/proper/commons-codec/apidocs/org/apache/commons/codec/binary/Base32.html
We can of course decide to use the platform line ending if we think this is silly...
But having the same file generated independent of the platform also has its merits.

Reply:
The rfc seems to say:

> Implementations MUST NOT add line feeds to base-encoded data unless
   the specification referring to this document explicitly directs base
   encoders to add line feeds after a specific number of characters.


So I would not want to default to newline split output without a way to turn that off.

Reply:
After some more discussion, removed the `lineWidth` keyword parameter so the line endings are no longer relevant.